sdk/core/azure-core/inc/azure/core/http/curl/curl connection_pool.hpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief The curl connection pool provides the utilities for creating a new curl connection and to
+ * keep a pool of connections to be re-used.
+ */
+
+#pragma once
+
+#include "azure/core/http/curl/curl_connection.hpp"
+#include "azure/core/http/http.hpp"
+
+#include <curl/curl.h>
+#include <list>
+#include <map>
+#include <memory>
+#include <mutex>
+
+#ifdef TESTING_BUILD
+// Define the class name that reads from ConnectionPool private members
+namespace Azure { namespace Core { namespace Test {
+  class TransportAdapter_connectionPoolTest_Test;
```

Comment:
The name you selected doesn't meet our guidelines.
```suggestion
  class ConnectionPoolTest;
```

Reply:
It gets generated by gtest.
The name is only `connectionPoolTest`. gtest adds the class name and the label `Test` at the end